doc/user-manual/language/reflection.lagda.rst

```diff
@@ -711,3 +711,36 @@ Example usage:
       defId id-name
 
     unquoteDecl id′ = mkId id′
+
+System Calls
+~~~~~~~~~~~~
+
+It is possible to run system calls as part of a metaprogram, using the ``execTC`` builtin. You can use this feature to implement type providers, or to call external solvers. For instance, the following example calls ``/bin/echo`` from Agda:
+
+.. code-block:: agda
+
+  postulate
+    execTC : (exe : String) (args : List String) (stdIn : String)
+           → TC (Σ Nat (λ _ → Σ String (λ _ → String)))
+
+  {-# BUILTIN AGDATCMEXEC execTC #-}
+
+  macro
```

Comment:
Can this code be placed as a test case in `test/Succeed/`?

Reply:
The problem with a successful test is that I don't know which executables are available when the tests are run. I don't think `echo` is available cross-platform, so this would always be a failing test on Windows.

I believe we discussed this during the meeting, and arrived at the fact that we don't really know of any executable which is available on all platforms, save perhaps `agda`, so we could run `agda --version` as a test, and see that it returns the current version? However, I'm not sure how to grab the path for Agda from a test, or how to setup the `.agda/executables` file for the test...